zigen-project/zen
# move/resize by client-given cursor texture

## Context

implement changing the cursor texture, move / resize the view, rebase event

## Summary

- 228ee3e8956b4877ec7edcdd9515fae89c62c9e5 : Listen set_cursor signal, update cursor texture when the client gives the cursor texture
- b5495d90b6bd35282c203c32c72ea2e072ba048b : make possible to move the view
- cdc1fbfe02b0b746ae9473a70d10759587a85c48 : make possible to resize the view
- ea14dd779d56657527a0be062aa35ed4ddf6e1bb : implement `rebase` to cursor grab
  - fix bug that moving was not able when move had just finished
- ea14dd779d56657527a0be062aa35ed4ddf6e1bb : focus and restack the view
- a0383cfcf4a309305176f69f45b399e9d07fa28a : change initial view pos to center

## How to check behavior

1. start zen with some client - the view is at the center of the board, and focused
2. move cursor onto the client surface; the cursor is may changed
  1. you can check this behavior with `weston-terminal`
4. try to move / resize the client
5. launch another client - the view is mapped to the forefront
